src/dune_memory/dune_memory.ml

```diff
@@ -96,6 +102,52 @@ let apply ~f o v =
   | Some o -> f v o
   | None -> v
 
+module MetadataFile = struct
+  type t =
+    { metadata : Sexp.t list
+    ; files : File.t list
+    }
+
+  let to_sexp { metadata; files } =
+    let open Sexp in
+    let f ({ in_the_build_directory; in_the_memory; _ } : File.t) =
+      Sexp.List
+        [ Sexp.Atom
+            (Path.Local.to_string (Path.Build.local in_the_build_directory))
+        ; Sexp.Atom (Path.to_string in_the_memory)
+        ]
+    in
+    List
+      [ List (Atom "metadata" :: metadata)
+      ; List (Atom "files" :: List.map ~f files)
+      ]
+
+  let of_sexp = function
+    | Sexp.List
+        [ List (Atom "metadata" :: metadata); List (Atom "files" :: produced) ]
+      ->
+      let+ files =
+        Result.List.map produced ~f:(function
+          | List [ Atom in_the_build_directory; Atom in_the_memory ] ->
+            let in_the_build_directory =
+              Path.Build.of_string in_the_build_directory
+            and in_the_memory = Path.of_string in_the_memory in
```

Comment:
Is this `Path.of_string` safe here?

Reply:
Yes, it's an absolute path. This is old code (like minimum 3 weeks 🐒), it just got moved around.

Reply:
Okay, then perhaps we should use `Path.External.of_string` then?